### PR TITLE
added command for send weekly emails

### PR DESCRIPTION
**Pull request by muak50**

### CODE REVIEW

File: common/djangoapps/philu_commands/management/commands/send_ondemand_weekly_emails.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+                        qualifiers={'category': 'course'}
+                    )
+                    if not course_chapters:
+                        log.info('%s Course don\'t have any chapters', course.display_name)
+                        continue
+
+                    # To maintain counter of graded sub section.
+                    graded_subsections = 0
+
+                    chapter = course_chapters[0].children[current_module - 2]
```
Comment (emraan004) [inline common/djangoapps/philu_commands/management/commands/send_ondemand_weekly_emails.py]: let's make a constant and write its purpose

File: common/djangoapps/philu_commands/management/commands/send_ondemand_weekly_emails.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+            enrollments = CourseEnrollment.objects.filter(course_id=course.id, is_active=True)
+
+            for enrollment in enrollments:
+                delta_days = today - enrollment.created.date()
+
+                # one is added to get user's current module.
+                current_module = int(math.floor((delta_days.days / DAYS_TO_COMPLETE_ONE_MODULE)) + 1)
+                user = enrollment.user
+
+                if not get_user_email_preferences(user, course.id):
```
Comment (ahmed-faraz46) [inline common/djangoapps/philu_commands/management/commands/send_ondemand_weekly_emails.py]: Please add a comment to describe the purpose of this sanity.

File: common/djangoapps/philu_commands/management/commands/send_ondemand_weekly_emails.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+                    # To maintain counter of graded sub section.
+                    graded_subsections = 0
+
+                    # We are subtracting 2 from current module, 1 for adjusting index as list indexs starts from 0 and
+                    # current module starts from 1 and other 1 to get previous module.
+                    chapter = course_chapters[0].children[current_module - 2]
+                    sequentials = modulestore().get_item(chapter)
+                    for index_sequential, sequential in enumerate(sequentials.children):
+                        if course_blocks[str(sequential)]['graded']:
+                            graded_subsections = graded_subsections + 1
```
Comment (ahmed-faraz46) [inline common/djangoapps/philu_commands/management/commands/send_ondemand_weekly_emails.py]: `+=`

File: common/djangoapps/philu_commands/management/commands/send_ondemand_weekly_emails.py [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+        anonymous_user = AnonymousUserId.objects.get(user=user, course_id=course_id)
+    except AnonymousUserId.DoesNotExist:
+        log.error('Anonymous Id doesn\'t exists for User: %s', user)
+    except AnonymousUserId.MultipleObjectsReturned:
+        log.error('Multiple Anonymous Ids for User: %s', user)
+    return anonymous_user
+
+
+def get_user_email_preferences(user, course_id):
+    email_pref = OnDemandEmailPreferences()
```
Comment (ahmed-faraz46) [inline common/djangoapps/philu_commands/management/commands/send_ondemand_weekly_emails.py]: Let's please delete this variable and use try, except to return.

**Comment by muak50** [comment]
Further optimization will be done later.

**Pull request merged**